src/GridFS/GridFsDownload.php

```diff
@@ -0,0 +1,148 @@
+<?php
+namespace MongoDB\GridFS;
+
+use MongoDB\Collection;
+use MongoDB\Exception\RuntimeException;
+use MongoDB\BSON\ObjectId;
+/**
+ * GridFsupload abstracts the processes of inserting into a GridFSBucket
+ *
+ * @api
+ */
+class GridFsDownload extends GridFsStream
+{
+    private $chunksIterator;
+    private $bytesSeen=0;
+    private $numChunks;
+    private $iteratorEmpty=false;
+    private $firstCheck=true;
+    private $bufferFresh=true;
+    private $bufferEmpty=true;
+    /**
+     * Constructs a GridFS download stream
+     *
+     * Supported options:
+     *
+     *  * contentType (string): DEPRECATED content type to be stored with the file.
+     *    This information should now be added to the metadata
+     *
+     *  * aliases (array of strings): DEPRECATED An array of aliases.
+     *    Applications wishing to store aliases should add an aliases field to the
+     *    metadata document instead.
+     *
+     *  * metadata (array or object): User data for the 'metadata' field of the files
+     *    collection document.
+     *
+     *  * writeConcern (MongoDB\Driver\WriteConcern): Write concern.
+     *
+     * @param array                  $options          File options
+     * @throws FileNotFoundException
+     */
+    public function __construct(
+        Bucket $bucket,
```

Comment:
Similar to https://github.com/mongodb/mongo-php-library/pull/57/files#r48283440, this class only needs the Bucket to do queries on the files and chunks collections. We should replace this argument with the Manager and GridFS prefix. If the read preference option is necessary, then we can inject both Collection instances instead.